### PR TITLE
Replace '.points.' with just '.'

### PR DESCRIPTION
**Pull request by kunaltyagi**

### CODE REVIEW

**Comment by kunaltyagi** [comment]
Depends on #4220

**Comment by kunaltyagi** [comment]
Ignore the changes affecting `size`. They are being handled more specifically in the separate PR.

File: common/include/pcl/point_cloud.h [hunk 1]
```diff
@@ -456,6 +456,7 @@ namespace pcl
       inline const_reverse_iterator crend () const noexcept { return (points.crend ()); }
 
       //capacity
+      index_t max_size() const noexcept { return static_cast<index_t>(points.max_size()); }
```
[the remark this answers is not in the document]
Comment (kunaltyagi) [reply common/include/pcl/point_cloud.h:459]: Sorry, forgot about this. I had created PRs for other members that I added, but forgot about this one.

**Comment by SergioRAgostinho** [comment]
Marking this as needs more work.

**Review by kunaltyagi — COMMENTED** [review]
Can't reproduce test failure in marching cubes.

File: test/gpu/octree/perfomance.cpp [hunk 2]
```diff
@@ -133,7 +133,7 @@ TEST(PCL_OctreeGPU, performance)
     cv::Octree octree_opencv;
     const static int opencv_octree_points_per_leaf = 32;    
     std::vector<cv::Point3f> opencv_points(data.size());
-    std::transform(data.points.begin(), data.points.end(), opencv_points.begin(), DataGenerator::ConvPoint<cv::Point3f>());
+    std::transform(data.begin(), data.end(), opencv_points.begin(), DataGenerator::ConvPoint<cv::Point3f>());
```
Comment (SergioRAgostinho) [inline test/gpu/octree/perfomance.cpp:136]: cbegin, cend

**Review by SergioRAgostinho — COMMENTED** [review]
The tests are complaining in all 3 OSes.

**Comment by kunaltyagi** [comment]
That was my bad. Please check now

**Comment by SergioRAgostinho** [comment]
> Can't reproduce test failure in marching cubes.

Not even in docker? We can't merge this until we understand what caused it. If we manage to isolate part of the code which has no influence we can merge it in advance but otherwise we're merging something which turns our tests red from now on and that doesn't make much sense.

File: surface/include/pcl/surface/impl/marching_cubes.hpp [hunk 3]
```diff
@@ -246,7 +246,7 @@ pcl::MarchingCubes<PointNT>::performReconstruction (pcl::PointCloud<PointNT> &po
   voxelizeData ();
 
   // preallocate memory assuming a hull. suppose 6 point per voxel
-  double size_reserve = std::min((double) intermediate_cloud.points.max_size (),
+  double size_reserve = std::min((double) intermediate_cloud.max_size (),
```
Comment (kunaltyagi) [inline surface/include/pcl/surface/impl/marching_cubes.hpp:249]: This is the line causing error. I've seen a similar fix (by @mvieth I think) regarding `std::min` or `std::max` where using them introduced an error.

The testing was done here:
https://kunaltyagi.visualstudio.com/pcl/_build?definitionId=3&_a=summary&repositoryFilter=2&branchFilter=99%2C99%2C99%2C99%2C99%2C99

Comment (mvieth) [reply surface/include/pcl/surface/impl/marching_cubes.hpp:249]: Not sure what fix you are referring to, but something like this seems better: `const std::size_t size_reserve = std::min(intermediate_cloud.max_size (), 2 * 6 * (std::size_t) (res_y_*res_z_ + res_x_*res_z_ + res_x_*res_y_));`

Comment (SergioRAgostinho) [reply surface/include/pcl/surface/impl/marching_cubes.hpp:249]: Isolate this change to another PR so this once can be merged.

**Review by SergioRAgostinho — APPROVED** [review]

**Pull request merged**